File: backend/src/models/Category.ts
import { ObjectId } from 'mongodb';

export interface Category {
  _id?: ObjectId;
  name: string;
  description: string;
  slug: string;
  order: number;
  isActive: boolean;
  createdAt: Date;
  updatedAt: Date;
  createdBy?: ObjectId; // Reference to User
} 

File: backend_spring/src/main/java/ch/wiss/forum/model/Category.java
package ch.wiss.forum.model;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "categories")
public class Category {
    
    @Id
    private String id;
    
    private String name;
    
    private String description;
    
    @Indexed(unique = true)
    private String slug;
    
    private int order;
    
    private boolean isActive;
    
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @DBRef
    private User createdBy;
} 